### PR TITLE
test: add file upload test case with popup handling

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright/issues/29923

### CODE REVIEW

File: tests/page/page-set-input-files.spec.ts [hunk 1]
```diff
@@ -37,6 +37,24 @@ it('should upload the file', async ({ page, server, asset }) => {
...
+  it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/29923' });
+  it.fixme(browserName === 'firefox');
+  await page.goto(server.PREFIX + '/input/fileupload.html');
+  {
+    const [popup] = await Promise.all([
+      page.waitForEvent('popup'),
+      page.evaluate(() => window['__popup'] = window.open('about:blank')),
+    ]);
+    await popup.close();
+    await page.waitForTimeout(1000)
```
Comment (mxschmitt) [inline tests/page/page-set-input-files.spec.ts:50]:
```suggestion
```

Comment (pavelfeldman) [inline tests/page/page-set-input-files.spec.ts:50]: mind the timeout.

File: tests/page/page-set-input-files.spec.ts [hunk 2]
```diff
@@ -37,6 +37,24 @@ it('should upload the file', async ({ page, server, asset }) => {
   }, input)).toBe('contents of the file');
 });
 
+it('should upload the file with a popup which opened/closed before', async ({ page, server, asset, browserName }) => {
```
Comment (mxschmitt) [inline tests/page/page-set-input-files.spec.ts:40]:
```suggestion
it('should upload the file after popup', async ({ page, server, asset, browserName }) => {
```

File: tests/page/page-set-input-files.spec.ts [hunk 3]
```diff
@@ -37,6 +37,23 @@ it('should upload the file', async ({ page, server, asset }) => {
   }, input)).toBe('contents of the file');
 });
 
+it('should upload the file after popup', async ({ page, server, asset, browserName }) => {
```
Comment (mxschmitt) [inline tests/page/page-set-input-files.spec.ts:40]:
```suggestion
it('should upload a file after popup', async ({ page, server, asset, browserName }) => {
```

**Pull request merged**